packages/react/src/hooks/useStorage.tsx

```diff
@@ -0,0 +1,19 @@
+import { S3ProviderGetConfig, Storage } from '@aws-amplify/storage';
+import { useEffect, useState } from 'react';
+
+export const useStorageURL = (key: string, options: S3ProviderGetConfig) => {
+  const [url, setURL] = useState<string>();
+  const [error, setError] = useState<Error>();
+
+  const fetch = () => {
+    Storage.get(key, options).then(setURL).catch(setError);
+  };
+
+  useEffect(fetch, []);
```

Comment:
When components are unmounted, these requests can be cancelled:
> https://docs.amplify.aws/lib/storage/cancel-requests/q/platform/js/

Reply:
resolved! thanks for the suggestion <3